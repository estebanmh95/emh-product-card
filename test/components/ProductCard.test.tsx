import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { ProductCard } from '../../src/components';
import { product1 } from '../data/products';
describe('it', () => {
  it('renders without crashing', () => {
    const wrapperTest = renderer.create(
      <ProductCard product={product1}>
        {() => <h1>Product Card</h1>}
      </ProductCard>
    );

    expect(wrapperTest.toJSON()).toMatchSnapshot();
  });

  it('Debe incrementar el contador', () => {
    const wrapperTest = renderer.create(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1, product1)}>Sumar</button>
          </>
        )}
      </ProductCard>
    );
    let tree = wrapperTest.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Debe incrementar el contador', () => {
    render(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1, product1)}>Sumar</button>
          </>
        )}
      </ProductCard>
    );

    const button = screen.getByRole('button', { name: 'Sumar' });

    expect(button).toBeInTheDocument();

    const preCounter = screen.queryByText(1);

    expect(preCounter).not.toBeInTheDocument();

    fireEvent.click(button);

    const counter = screen.getByText(1);

    expect(counter).toBeInTheDocument();
  });

  //   it('debe de mostrar el componente con el nombre del producto', () => {
  //     const wrapperTest = renderer.create(
  //       <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
  //     );

  //     expect(wrapperTest.toJSON()).toMatchSnapshot();
  //   });
});
