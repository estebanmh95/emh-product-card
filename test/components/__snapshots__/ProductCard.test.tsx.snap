// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it Debe incrementar el contador 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    Sumar
  </button>
</div>
`;

exports[`it renders without crashing 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;
